framework/src/utils/MortarUtils.C

```diff
@@ -222,11 +222,22 @@ projectQPoints3d(const Elem * const msm_elem,
         x1 += Moose::fe_lagrange_2D_shape(sub_element_type(), FIRST, n, xi) *
               primal_elem->point(sub_elem_node_indices[n]);
       auto u = x1 - x0;
+
       VectorValue<Dual2> F(u(1) * normal(2) - u(2) * normal(1),
                            u(2) * normal(0) - u(0) * normal(2),
                            u(0) * normal(1) - u(1) * normal(0));
 
-      if (MetaPhysicL::raw_value(F).norm() < 1e-12)
+      Real projection_tolerance(1e-10);
+
+      // Normalize tolerance with quantities involved in the projection.
+      // Projection tolerance is reduced for displacements larger than those on the order of one.
+      // Tightening the tolerance for displacements of smaller orders causes this tolerance to not
+      // be reached in a number of tests.
+      u(0) += 1.0e-36;
+      if (u.norm().value() > 1.0)
```

Comment:
```suggestion
      if (!u.is_zero() && u.norm().value() > 1.0)
```

Reply:
Ok instead of posing verbose rhetorical questions, I should actually just review more carefully to understand what's going on ... I think this is a little bit more clean and addresses my philosophical concern

Reply:
I should have searched the API more! In other cases we are adding the delta to a, e.g. std::sqrt, which is a different case, since we need the value for the residual.